File: benchmarks/vm/10_match_loop.lua
local t0 = os.clock()
local total = 0
for i = 0, 199999 do
    local r = i % 5
    if r == 0 then
        total = total + i
    elseif r == 1 then
        total = total - i
    elseif r == 2 then
        total = total + (i * 2)
    else
        total = total + 1
    end
end
local total_ms = (os.clock() - t0) * 1000
print("total = " .. tostring(total))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))

File: benchmarks/vm/10_match_loop.py
import time

t0 = time.monotonic()
total = 0
for i in range(200000):
    match i % 5:
        case 0:
            total += i
        case 1:
            total -= i
        case 2:
            total += i * 2
        case _:
            total += 1
total_ms = (time.monotonic() - t0) * 1000
print(f"total = {total}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")
